salt/scripts.py

```diff
@@ -69,17 +45,25 @@ def _handle_signals(client, signum, sigframe):
     else:
         exit_msg = None
 
-    _handle_interrupt(
-        SystemExit(exit_msg),
-        Exception("\nExiting with hard crash on Ctrl-c"),
-        hardcrash,
-        trace=trace,
-    )
+    if exit_msg is None and hardcrash:
+        exit_msg = "\nExiting with hard crash on Ctrl-c"
+    if exit_msg:
+        print(exit_msg, file=sys.stderr, flush=True)
+    if hardcrash:
+        try:
+            # This raises AttributeError on Python 3.4 and 3.5 if there is no current exception.
```

Comment:
**suggestion (non-blocking)** we should mark when to remove this. I forget when we completely drop 3.5 support, but... after that point :stuck_out_tongue_closed_eyes: